parsing/parchmint-parser.js

```diff
@@ -372,9 +362,17 @@ class ParchmintParser {
      * @returns {Connection}    The resulting Connection object.
      */
     getParsedConnection(connObj) {
-        return new Connection(connObj['name'], connObj['id'], this.getParsedTerminal(connObj['source'],
-                connObj['layer']), this.getParsedTerminals(connObj['sinks'], connObj['layer']),
-                this.connFeatures.get(connObj['layer']));
+        let ret = new Connection(connObj['name'], connObj['id'], this.getParsedTerminal(connObj['source'],
+                connObj['layer']), this.getParsedTerminals(connObj['sinks'], connObj['layer']));
+        let feature = this.connFeatures.get(connObj['layer']);
+
+        // Connection Features are not required, so we only add them to the Connection if we parsed some,
+        // otherwise we'll leave it null.
+        if (feature) {
+            ret.segments = feature;
+        }
+
+        return ret;
```

Comment:
Didn't we decide in the last PR that this functionality should be done directly in the constructor rather than externally? Perhaps this was an error in the merge?

Reply:
Nearly every time it's merged since we added that in it changes to this without any conflicts. I missed it this time. It's been fixed.